test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -0,0 +1,275 @@
+                                    new Email("bob@nusgreyhats.org", false),
+                                    new Address("9 Computing Drive", false),
+                                    new UniqueTagList(tagMathematician));
+
+        charlieDouglas = new Person(new Name("Charlie Douglas"),
+                                    new Phone("98751365", false),
+                                    new Email("charlie@nusgdg.org", false),
+                                    new Address("10 Science Drive", false),
+                                    new UniqueTagList(tagScientist));
+
+        davidElliot    = new Person(new Name("David Elliot"),
+                                    new Phone("84512575", false),
+                                    new Email("douglas@nuscomputing.com", false),
+                                    new Address("11 Arts Link", false),
+                                    new UniqueTagList(tagEconomist, tagPrizeWinner));
+    }
+    
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void syncTagsWithMasterList_emptyInitialTagList_addsAllIntoTagList() throws Exception {
+        UniqueTagList emptyTagList = new UniqueTagList();
+        UniquePersonList persons = new UniquePersonList(bobChaplin, charlieDouglas);
+
+        AddressBook addressBook = new AddressBook(persons, emptyTagList);
+
+        assertTrue(addressBook.containsTag(tagMathematician));
+        assertTrue(addressBook.containsTag(tagScientist));
+        assertFalse(addressBook.containsTag(tagEconomist));
+
+    }
+
+    @Test
+    public void syncTagsWithMasterList_someTagsNotInTagList_addsMissingTagsIntoTagList() throws Exception {
+        AddressBook addressBook = createDefaultAddressBook();
+
+        assertFalse(addressBook.containsTag(tagEconomist));
+        assertFalse(addressBook.containsTag(tagPrizeWinner));
+        addressBook.addPerson(davidElliot);
+        assertTrue(addressBook.containsTag(tagEconomist));
+        assertTrue(addressBook.containsTag(tagPrizeWinner));
+    }
+
+    @Test
+    public void syncTagsWithMasterList_defaultAddressBook_tagEqualityPropertyHolds() throws Exception {
+        AddressBook addressBook = createDefaultAddressBook();
+        UniquePersonList persons = addressBook.getAllPersons();
+        UniqueTagList tagsInPersons = getAllTags(persons);
+        UniqueTagList tags = addressBook.getAllTags();
+        
+        // all person tags are equal to some tag in the master list
+        boolean allPersonTagsValid = true;
+        for (Tag personTag : tagsInPersons) {
+            // particular person tag is equal to some tag in the master list
+            boolean personTagValid = false;
+            for (Tag tag : tags) {
+                personTagValid = personTagValid || (personTag == tag);
+            }
+            allPersonTagsValid = allPersonTagsValid && personTagValid;
+        }
+        
+        assertTrue(allPersonTagsValid);
+    }
```

Comment:
What does this test case tests that the previous methods don't?
I think the only functionality we can test is when a person is added, if that person has a tag that is missing in the tag list, it will be added to the list. In fact, it can be tested while testing `addPerson` method.